src/Altinn.Platform/Altinn.Platform.Storage/Storage/Helpers/AuthorizationHelper.cs

```diff
@@ -0,0 +1,299 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Altinn.Authorization.ABAC.Xacml.JsonProfile;
+using Altinn.Common.PEP.Constants;
+using Altinn.Common.PEP.Helpers;
+using Altinn.Common.PEP.Interfaces;
+using Altinn.Platform.Storage.Interface.Models;
+
+namespace Altinn.Platform.Storage.Helpers
+{
+    /// <summary>
+    /// Custom PEP to create multi decision request.
+    /// </summary>
+    public class AuthorizationHelper
+    {
+        private readonly IPDP _pdp;
+
+        private const string XacmlResourceTaskId = "urn:altinn:task";
+        private const string XacmlResourceEndId = "urn:altinn:end-event";
+        private const string XacmlResourceActionId = "urn:oasis:names:tc:xacml:1.0:action:action-id";
+        private const string DefaultIssuer = "Altinn";
+        private const string DefaultType = "string";
+        private const string SubjectId = "s";
+        private const string ActionId = "a";
+        private const string ResourceId = "r";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthorizeInstancesHelper"/> class.
+        /// </summary>
+        /// <param name="pdp">The policy decision point</param>
+        public AuthorizationHelper(IPDP pdp)
+        {
+            _pdp = pdp;
+        }
+
+        /// <summary>
+        /// Creates multi decision request.
+        /// </summary>
+        public static XacmlJsonRequestRoot CreateMultiDecisionRequest(ClaimsPrincipal user, List<Instance> instances, List<string> actionTypes)
+        {
+            if (user == null)
+            {
+                throw new ArgumentNullException("user");
+            }
+
+            XacmlJsonRequest request = new XacmlJsonRequest();
+            request.AccessSubject = new List<XacmlJsonCategory>();
+            request.AccessSubject.Add(CreateMultipleSubjectCategory(user.Claims));
+            request.Action = CreateMultipleActionCategory(actionTypes);
+            request.Resource = CreateMultipleResourceCategory(instances);
+            request.MultiRequests = CreateMultiRequestsCategory(request.AccessSubject, request.Action, request.Resource);
+
+            XacmlJsonRequestRoot jsonRequest = new XacmlJsonRequestRoot() { Request = request };
+
+            return jsonRequest;
+        }
+
+        private static XacmlJsonCategory CreateMultipleSubjectCategory(IEnumerable<Claim> claims)
```

Comment:
In my opinion I prefer public methods above private methods.

Reply:
Moved all public methods to the top